cmd/thanos/store.go

```diff
@@ -45,7 +45,7 @@ func registerStore(m map[string]setupFunc, app *kingpin.Application) {
 	httpBindAddr, httpGracePeriod := regHTTPFlags(cmd)
 	grpcBindAddr, grpcGracePeriod, grpcCert, grpcKey, grpcClientCA := regGRPCFlags(cmd)
 
-	dataDir := cmd.Flag("data-dir", "Data directory in which to cache remote blocks.").
+	dataDir := cmd.Flag("data-dir", "Data directory on the disk space for local cache which stores information about all the remote blocks.").
```

Comment:
Can we maybe expand a bit? (: More helpful flag message, better!

```suggestion
	dataDir := cmd.Flag("data-dir", "Local data directory used for caching purposes (index-header, in-mem cache items and meta.jsons). If removed, no data will be lost, just store will have to rebuild the cache. NOTE: Putting raw blocks here will not cause the store to read them. For such use cases use Prometheus + sidecar.").
```